test_unstructured_ingest/dest/pinecone.sh

```diff
@@ -0,0 +1,148 @@
+#!/usr/bin/env bash
+
+set -e
+
+DEST_PATH=$(dirname "$(realpath "$0")")
+SCRIPT_DIR=$(dirname "$DEST_PATH")
+cd "$SCRIPT_DIR"/.. || exit 1
+OUTPUT_FOLDER_NAME=s3-pinecone-dest
+OUTPUT_DIR=$SCRIPT_DIR/structured-output/$OUTPUT_FOLDER_NAME
+WORK_DIR=$SCRIPT_DIR/workdir/$OUTPUT_FOLDER_NAME
+DOWNLOAD_DIR=$SCRIPT_DIR/download/$OUTPUT_FOLDER_NAME
+max_processes=${MAX_PROCESSES:=$(python3 -c "import os; print(os.cpu_count())")}
+
+if [ -z "$PINECONE_API_KEY" ]; then
+   echo "Skipping Pinecone ingest test because PINECONE_API_KEY env var is not set."
+   exit 0
+fi
+
+RANDOM_SUFFIX=$((RANDOM % 100000 + 1))
+
+PINECONE_ENVIRONMENT="us-east1-gcp"
+PINECONE_INDEX="ingest-test-$RANDOM_SUFFIX"
+PINECONE_PROJECT_ID="art8iaj"
+
+# shellcheck disable=SC1091
+source "$SCRIPT_DIR"/cleanup.sh
+function cleanup {
+
+  # Get response code to check if index exists
+  response_code=$(curl \
+    -s -o /dev/null \
+    -w "%{http_code}" \
+    --request GET \
```

Comment:
Can a `HEAD` request suffice if all we're doing is checking for existence via the response code?

Reply:
This one gives a 405 method not allowed

Reply:
However this inquiry caused me to update a log by: [update log](https://github.com/Unstructured-IO/unstructured/pull/1774/commits/fa083ff7c08da6b72f869a3c4d9ffc4ae2ccbdfa)